include/clang/Sema/AvailableFactsAnalysis.h

```diff
@@ -0,0 +1,82 @@
+//===-------- AvailableFactsAnalysis.h - collect comparison facts --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file defines the interface for performing an analysis for collecting 
+//  comparison facts.
+//
+//  The analysis has the following characteristics: 1. forward dataflow analysis,
+//  2. conservative, 3. intra-procedural, and 4. path-sensitive.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_AVAILABLE_FACTS_ANALYSIS_H
+#define LLVM_AVAILABLE_FACTS_ANALYSIS_H
+
+#include "clang/Analysis/CFG.h"
+#include "clang/Analysis/Analyses/PostOrderCFGView.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "clang/Sema/Sema.h"
+#include <queue>
+
+namespace clang {
+  typedef std::set<const Expr *> ExprSet;
+  typedef std::pair<Expr *, Expr *> Comparison;
```

Comment:
You should use llvm containers like SmallPtrSet, SmallSet, etc instead of std ones.